File: 17. Letter Combinations of a Phone Number/Solution.py
class Solution:
    def gen(self,strr: List[str],res: List[str],cur: str,ind: int)->None:
        if ind==len(strr):
            res.append(cur)
            return
        for i in strr[ind]:
            self.gen(strr,res,cur+i,ind+1)
    def letterCombinations(self, digits: str) -> List[str]:
        strr,res = [],[]
        for i in digits:
            if i=='2': strr.append("abc")
            if i=='3': strr.append("def")
            if i=='4': strr.append("ghi")
            if i=='5': strr.append("jkl")
            if i=='6': strr.append("mno")
            if i=='7': strr.append("pqrs")
            if i=='8': strr.append("tuv")
            if i=='9': strr.append("wxyz")
        if len(strr)==0: return res
        self.gen(strr,res,"",0)
        return res

File: 17. Letter Combinations of a Phone Number/Solution.cpp
class Solution {
public:
    void gen(vector<string> &str, vector<string> &res, string current, int ind){
        if(ind==str.size()){
            res.push_back(current);
            return;
        }
        for(char chh : str[ind]){
            gen(str,res,current+chh,ind+1);
        }
    }
    vector<string> letterCombinations(string digits) {
        vector<string> str,res;
        for(char ch : digits){
            if(ch=='2'){str.push_back("abc");}
            if(ch=='3'){str.push_back("def");}
            if(ch=='4'){str.push_back("ghi");}
            if(ch=='5'){str.push_back("jkl");}
            if(ch=='6'){str.push_back("mno");}
            if(ch=='7'){str.push_back("pqrs");}
            if(ch=='8'){str.push_back("tuv");}
            if(ch=='9'){str.push_back("wxyz");}
        }
        if(str.size()==0){
            return res;
        }
        gen(str,res,"",0);
        return res;
    }
};
